dao/src/main/java/org/thingsboard/server/dao/sql/JpaAbstractDao.java

```diff
@@ -51,6 +62,33 @@ protected void setSearchText(E entity) {
     @Override
     @Transactional
     public D save(TenantId tenantId, D domain) {
+        E entity = prepare(domain);
+        entity = getRepository().save(entity);
+        return DaoUtil.getData(entity);
+    }
+
+    @Override
+    @Transactional
+    public D saveAndFlush(TenantId tenantId, D domain) {
+        D d = save(tenantId, domain);
+        getRepository().flush();
+        return d;
+    }
+
+    @Override
+    public D create(TenantId tenantId, D domain) {
+        E entity = prepare(domain);
```

Comment:
On this stage we have a marker do we need to create or save:
```
if (entity.getUuid() == null) {
```
So if we have a new entity with null id - it should create. otherwise - save by default.
The interface might remain the same. Services will ask to save and we will find out how to handle in the internal implementation depending on the new or existing entity. 
It seems good for me as we use random UUID as an ID (effectively unique)

Reply:
For partitioned entities we preset the id sooner, and here we have the entity already with id.

Reply:
But I agree it would be good to refactor this for non-partitioned entities too (in the `save` call create in case the id is null).